src/it/java/org/checkstyle/plugins/sonar/RunPluginTest.java

```diff
@@ -0,0 +1,260 @@
+                .setAdminCredentials()
+                .setMethod(HttpMethod.POST)
+                .setParam("language", "java")
+                .setParam("name", "checkstyle")
+                .execute()
+                .getBodyAsString();
+        Map<String, Object> map = new Gson().fromJson(profilesResponse, Map.class);
+        String profileKey = ((Map<String, String>) map.get("profile")).get("key");
+        if(StringUtils.isEmpty(profileKey)) {
+            fail("Could not retrieve profile key: setting up quality profile failed.");
+        } else {
+            HttpResponse activateRulesResponse = orchestrator.getServer()
+                    .newHttpCall("api/qualityprofiles/activate_rules")
+                    .setAdminCredentials()
+                    .setMethod(HttpMethod.POST)
+                    .setParam("activation_severity", "INFO")
+                    .setParam("languages", "java")
+                    .setParam("profile_key", profileKey)
+                    .setParam("repositories", "checkstyle")
+                    .executeUnsafely();
+            if(!activateRulesResponse.isSuccessful()) {
+                fail(String.format("Failed to activate all rules. %s",
+                        activateRulesResponse.getBodyAsString()));
+            }
+            // deactivate some rules for test project
+            for(String ruleKey : DEACTIVATED_RULES) {
+                HttpResponse deactivateRulesResponse = orchestrator.getServer()
+                        .newHttpCall("api/qualityprofiles/deactivate_rule")
+                        .setAdminCredentials()
+                        .setMethod(HttpMethod.POST)
+                        .setParam("rule_key", "checkstyle:" + ruleKey)
+                        .setParam("profile_key", profileKey)
+                        .executeUnsafely();
+                if(!deactivateRulesResponse.isSuccessful()) {
+                    fail(String.format("Failed to deactivate rule %s. %s",
+                            ruleKey,
+                            deactivateRulesResponse.getBodyAsString()));
+                }
+            }
+        }
+
+        // associate CS profile
+        orchestrator.getServer().provisionProject(PROJECT_KEY, PROJECT_NAME);
+        HttpResponse assignQPResponse = orchestrator.getServer()
+                .newHttpCall("api/qualityprofiles/add_project")
+                .setAdminCredentials()
+                .setMethod(HttpMethod.POST)
+                .setParam("language", "java")
+                .setParam("profileName", "checkstyle")
+                .setParam("projectKey", PROJECT_KEY)
+                .executeUnsafely();
+        if(!assignQPResponse.isSuccessful()) {
+            fail(String.format("Failed to add project to quality profile. %s",
+                    assignQPResponse.getBodyAsString()));
+        }
+
+        // copy project to analysis space
+        Path projectRoot = Paths.get("test-resources/" + PROJECT_NAME);
```

Comment:
please move project that we need for ITs to `it` resources
`Path projectRoot = Paths.get("it/java/resources/" + PROJECT_NAME);`

Reply:
Does it make sense to move resources required for `it` under such folder ? We split tests by folders, the same is good to do for resources

Reply:
yes sure, i actually moved them already but it seems this commit got lost or i didn't finish committing it - sorry for that!
i moved them to `it/resources/`